### DISCUSSION TITLE
Adding custom property method to LogOptions class

### DISCUSSION THREAD

**Discussion by Lukerayn3r** [spatie/laravel-activitylog#944] [Q&A] [answered]
Is there anything in the pipeline for v4 where we will be able to set custom properties on model events? I understand this may interfere with the new `->changes()` method but it would be great to be able to push extra data into the `properties` column.

I have an edge case where I would love to add an `account_id` to all my logs allowing me to search for all the logs for a particular account.

I am able to use `withProperties()` when defining my own log but it would also be great to do this on my model events.

Any help/guidance will be greatly appreciated.

**Comment by Gummibeer** [comment, marked as answer]
Hey,

you can do this by adding a new custom column or add it to the `properties`. To do so you can use the `tap()` logic documented or add an observer and use the `creating` event to define additional values.

This is already discussed and documented several times in the issues and discussions - so I won't provide additional input.
